src/app/storage/duck/operations.ts

```diff
@@ -185,38 +186,36 @@ function fetchMigStorageRefs(client: IClusterClient, migMeta, migStorages): Arra
       CoreNamespacedResourceKind.Secret,
       secretRef.namespace
     );
-    refs.push(client.get(secretResource, secretRef.name));
+    const resp = client.get(secretResource, secretRef.name);
+    resp.then(r => {
+      refs.push(r.data);
+    }, err => {
+      console.error('No secret matches', err.message);
+    });
   });
 
   return refs;
 }
 
-function groupStorages(migStorages: any[], refs: any[]): any[] {
+function groupStorages(migStorages: any[], secrets: any[]): any[] {
   return migStorages.map(ms => {
-    const fullStorage = {
+    return {
       MigStorage: ms,
+      Secret: secrets.find(
+        s => s.metadata.name === ms.spec.backupStorageConfig.credsSecretRef.name
+      )
     };
-    fullStorage['Secret'] = refs.find(
-      i => i.data.kind === 'Secret' && i.data.metadata.name === ms.metadata.name
-    ).data;
-
-    return fullStorage;
   });
 }
 
 function* fetchStorageGenerator() {
   const state = yield select();
   const client: IClusterClient = ClientFactory.hostCluster(state);
   const resource = new MigResource(MigResourceKind.MigStorage, state.migMeta.namespace);
-  try {
-    let storageList = yield client.list(resource);
-    storageList = yield storageList.data.items;
-    const refs = yield Promise.all(fetchMigStorageRefs(client, state.migMeta, storageList));
-    const groupedStorages = groupStorages(storageList, refs);
-    return { updatedStorages: groupedStorages, isSuccessful: true };
-  } catch (e) {
-    return { e, isSuccessful: false };
-  }
+  let storageList = yield client.list(resource);
+  storageList = yield storageList.data.items;
+  const secretRefs = yield Promise.all(fetchMigStorageSecretRefs(client, state.migMeta, storageList));
```

Comment:
This logic is going to cause problems for the same reasons as above. It should use similar logic to the above function. In fact, the logic I described is likely going to be so close that it should be brought out into a shared function so as not to duplicate the code.